std/src/main/scala/algebra/std/map.scala

```diff
@@ -26,33 +50,80 @@ class MapEq[K, V](implicit ev: Eq[V], a: AdditiveMonoid[V]) extends Eq[Map[K, V]
     check(x, y) && check(y, x)
 }
 
-class MapRng[K, V](implicit ev: Rng[V]) extends Rng[Map[K, V]] {
+class MapVectorOrder[K, V](implicit K: Order[K], V: Order[V], V0: AdditiveMonoid[V]) extends Order[Map[K, V]] {
+  def compare(x: Map[K, V], y: Map[K, V]): Int = {
+    val zero = V0.zero
+    def check(progress: (K, Int), x: Map[K, V], y: Map[K, V], sign: Int): (K, Int) = {
+      val it = x.iterator
+      var mink: K = progress._1
+      var outv: Int = progress._2
+      while (it.hasNext) {
+        val (k, v1) = it.next
+        if (outv == 0 || K.lt(k, mink)) {
+          val v2 = y.getOrElse(k, zero)
+          val r = V.compare(v1, v2)
+          if (r != 0) {
+            mink = k
+            outv = r * sign
+          }
+        }
+      }
+      (mink, outv)
+    }
+
+    val init = (null.asInstanceOf[K], 0)
+    val progress = check(init, x, y, 1)
+    val (_, result) = check(progress, y, x, -1)
+    result
+  }
+}
+
+class MapMonoid[K, V](implicit sg: Semigroup[V]) extends Monoid[Map[K, V]]  {
+  def empty: Map[K, V] = Map.empty
+
+  def combine(x: Map[K, V], y: Map[K, V]): Map[K, V] =
+    StaticMethods.combineMap(x, y)(sg.combine)(v => v)
+}
+
+class MapGroup[K, V](implicit g: Group[V]) extends MapMonoid[K, V] with Group[Map[K, V]] {
+  def inverse(x: Map[K, V]): Map[K, V] =
+    x.map { case (k, v) => (k, g.inverse(v)) }
```

Comment:
might be a case for the laziness of mapValues assuming we access each item on average <= 1 times.